Dynavity/Dynavity/view/CanvasElementMapView.swift

```diff
@@ -12,9 +12,16 @@ struct CanvasElementMapView: View {
 
     var body: some View {
         ZStack {
-            ForEach(elements, id: \.visualID) { element in
-                CanvasElementView(element: element)
-                    .offset(x: element.position.x, y: element.position.y)
+            ForEach(elements, id: \.id) { element in
+                // The group allows us to have common view modifiers.
```

Comment:
> By common modifier, you mean `.offset(x: element.position.x, y: element.position.y)`? i.e. you don't need to write `imageView.offset(...)` and `elementView.offset(...)`?

Yup!

> Would it be better to use a similar construct like so: https://stackoverflow.com/a/34800922, then just write offset once on the assigned view?

Could you elaborate further on this? The `Group` is there because the Swift compiler is not smart enough to determine that the output of the `if`-`else` block is guaranteed to be a `View`. Not too sure how that solves this issue.

Reply:
> Could you elaborate further on this? The Group is there because the Swift compiler is not smart enough to determine that the output of the if-else block is guaranteed to be a View. Not too sure how that solves this issue.

Whoops, you're right. My bad!